amethyst_ui/src/text.rs

```diff
@@ -35,133 +50,682 @@ impl UiText {
+                            text.get_mut(entity)
+                                .into_iter()
+                                .zip(editable.get_mut(entity).into_iter())
+                                .next()
                         });
+                        if let Some((ref mut focused_text, ref mut focused_edit)) =
+                            focused_text_edit 
+                        {
+                            use std::f32::NAN;
+
+                            let mouse_x = self.mouse_position.0;
+                            let mouse_y = self.mouse_position.1;
+                            // Find the glyph closest to the mouse position.
+                            focused_edit.highlight_vector = focused_text
+                                .cached_glyphs
+                                .iter()
+                                .enumerate()
+                                .fold((0, (NAN, NAN)), |(index, (x, y)), (i, g)| {
+                                    let pos = g.position();
+                                    // Use Pythagorean theorem to compute distance
+                                    if ((x - mouse_x).powi(2) + (y - mouse_y).powi(2)).sqrt()
+                                        < ((pos.x - mouse_x).powi(2)
+                                            + (pos.y - mouse_y).powi(2))
+                                            .sqrt()
+                                    {
+                                        (index, (x, y))
+                                    } else {
+                                        (i, (pos.x, pos.y))
+                                    }
+                                })
+                                .0
+                                as isize - focused_edit.cursor_position;
+                            // The end of the text, while not a glyph, is still something
+                            // you'll likely want to click your cursor to, so if the cursor is
+                            // near the end of the text, check if we should put it at the end
+                            // of the text.
+                            if focused_edit.cursor_position + focused_edit.highlight_vector + 1
+                                == focused_text.cached_glyphs.len() as isize
+                            {
+                                if let Some(last_glyph) =
+                                    focused_text.cached_glyphs.iter().last()
+                                {
+                                    if (last_glyph.position().x - mouse_x).abs()
+                                        > ((last_glyph.position().x
+                                            + last_glyph
+                                                .unpositioned()
+                                                .h_metrics()
+                                                .advance_width)
+                                            - mouse_x)
+                                            .abs()
```

Comment:
This is some weird formatting...

Reply:
rustfmt-nightly 0.2.16 official.  But yeah it does look odd.